wolfcrypt/src/port/nxp/se050_port.c

```diff
@@ -0,0 +1,1418 @@
+            status = sss_symmetric_context_init(&aes->aes_ctx, cfg_se050_i2c_pi, &aes->AesKeyCtx, algorithm, mode);
+            if (status != kStatus_SSS_Success) {
+                printf("error 1 - aes_crypt\n");
+                return -1;
+            }
+        
+            status = sss_cipher_init(&aes->aes_ctx, (uint8_t *)aes->reg, sizeof(aes->reg));
+            if (status == kStatus_SSS_InvalidArgument)
+                printf("invalid arg\n");
+            if (status == kStatus_SSS_Fail)
+                printf("plain fail\n");
+            if (status != kStatus_SSS_Success) {
+                printf("error 3 - aes_crypt\n");
+                return -1;
+            }
+        }
+        status = sss_cipher_update(&aes->aes_ctx, in, sz, out, &sz);
+        if (status == kStatus_SSS_InvalidArgument)
+            printf("invalid arg\n");
+        if (status == kStatus_SSS_Fail)
+            printf("plain fail\n");
+        if (status != kStatus_SSS_Success) {
+            printf("error 4 - aes_crypt\n");
+            return -1;
+        }
+        //sss_symmetric_context_free(&aes_ctx); //matched up to aes free now
+    }
+
+    wolfSSL_CryptHwMutexUnLock();
+
+    return 0;
+
+}
+
+/*int se050_aes_ctr_crypt(Aes* aes, const byte* in, byte* out, word32 sz)
+{
+    //printf("running aes_ctr_crypt inside se050 port\n");
+    sss_status_t status;
+    byte* tmp;
+    //XMEMSET(&tmp, 0, sizeof(tmp));
+    //printf("0 - aes->left = %d\n", aes->left);
+    tmp = (byte*)aes->tmp + AES_BLOCK_SIZE - aes->left;
+    while (aes->left && sz) {
+      //  printf("got inside top while loop\n");
+       *(out++) = *(in++) ^ *(tmp++);
+       aes->left--;
+       sz--;
+    }
+    //printf("1 - aes->left = %d\n", aes->left);
+    
+
+    if (wolfSSL_CryptHwMutexLock() == 0)  {
+         if (aes->ctxInitDone == 0) {
+            aes->ctxInitDone = 1;
+            //printf("running context init\n");
+            status = sss_symmetric_context_init(&aes->aes_ctx, cfg_se050_i2c_pi,
+                                                &aes->AesKeyCtx, kAlgorithm_SSS_AES_CTR,
+                                                kMode_SSS_Encrypt);
+            if (status != kStatus_SSS_Success) {
+                printf("error 1 - aes_crypt\n");
+                return -1;
+            }
+        }
+        uint8_t  *iv;
+        iv = (uint8_t *)aes->reg;
+
+       // printf("running crypt_ctr function\n");
+        status = sss_cipher_crypt_ctr(&aes->aes_ctx, in, out, sz, iv, (uint8_t *)aes->tmp, (size_t *)&aes->left);
+        if (status == kStatus_SSS_InvalidArgument)
+            printf("invalid arg\n");
+        if (status == kStatus_SSS_Fail)
+            printf("plain fail\n");
+        if (status != kStatus_SSS_Success) {
+            printf("error 4 - aes_crypt\n");
+            return -1;
+        }
+    }
+    wolfSSL_CryptHwMutexUnLock();
+    //printf("ran through se050_ctr without errors...\n");
+    return 0;
+}*/
+
+struct ecc_key;
```

Comment:
Is this required? Seems like a workaround. All of the header includes and pre-def's should be at top of file.